### PR TITLE
Make negative values for gpuarray.preallocate disable the allocation cache completely.

### PR DESCRIPTION
**Pull request by abergeron**

### CODE REVIEW

**Comment by lamblin** [comment]
Can you add documentation for that flag to `doc/library/config.txt`?

**Comment by abergeron** [comment]
Rebased for the move of gpuarray.

**Comment by nouiz** [comment]
At the same time, can you add the printing of the value of the cache, like we do for cnmem?

File: theano/gpuarray/__init__.py [hunk 1]
```diff
@@ -55,37 +55,39 @@ def init_dev(dev, name=None):
                          disable_alloc_cache=config.gpuarray.preallocate < 0)
         init_dev.devmap[dev] = ctx
         if config.gpuarray.preallocate > 0:
+            MB = (1024 * 1024)
             if config.gpuarray.preallocate < 1:
```
Comment (nouiz) [inline theano/gpuarray/__init__.py]: Should be <=

**Comment by abergeron** [comment]
fixed.

**Comment by nouiz** [comment]
ok to merge after travis doc. I'm kiling the other travis part.

**Comment by abergeron** [comment]
I fixed the typo in the doc.

File: doc/library/config.txt [hunk 2]
```diff
@@ -442,6 +442,42 @@ import theano and print the config variable, as in:
...
+    done outside of debugging.
+
+        * < 0: disabled
+        * 0 <= N <= 1: use this fraction of the total GPU memory (clipped to .95 for driver memory).
+        * > 1: use this number in megabytes (MB) of memory.
+
+    .. note::
+
+        This could cause memory fragmentation. So if you have a
+        memory error while using CNMeM, try to allocate more memory at
```
Comment (lamblin) [inline doc/library/config.txt]: CNMeM?

Comment (abergeron) [inline doc/library/config.txt]: https://github.com/Theano/Theano/pull/4498

**Pull request merged**